packages/next/lib/verifyTypeScriptSetup.ts

```diff
@@ -0,0 +1,265 @@
+  );
+}
+
+async function verifyNoTypeScript(dir: string) {
+  const typescriptFiles = []
+  const firstDepth = (await readdir(dir)).filter(
+    (path) => path !== 'node_modules',
+  ).map((p) => path.join(dir, p))
+
+  for (const curPath of firstDepth) {
+    const info = await stat(curPath)
+
+    if (info.isFile() && !curPath.match(/.*\.d\.ts/)) {
+      typescriptFiles.push(curPath)
+    }
+    if (info.isDirectory()) {
+      const curFiles = (await recursiveReadDir(curPath, /.*\.ts/)).map((file) => path.join(curPath, file))
+      typescriptFiles.push(
+        ...curFiles.filter((file) => !file.match(/.*\.d\.ts/)),
+      )
+    }
+  }
+
+  if (typescriptFiles.length > 0) {
+    console.warn(
+      chalk.yellow(
+        `We detected TypeScript in your project (${chalk.bold(
+          `src${path.sep}${typescriptFiles[0]}`,
+        )}) and created a ${chalk.bold('tsconfig.json')} file for you.`,
+      ),
+    );
+    console.warn();
+    return false;
+  }
+  return true;
+}
+
+export default async function verifyTypeScriptSetup(dir: string) {
+  let firstTimeSetup = false;
+  const tsConfig = path.resolve(dir, 'tsconfig.json')
+  const yarnLockFile = path.resolve(dir, 'yarn.lock')
+  const nodeModules = (await findUp('node_modules', { cwd: dir }))!
+
+  if (!(await exists(tsConfig))) {
+    if (await verifyNoTypeScript(dir)) {
+      return;
+    }
+    await writeJson(tsConfig, {});
+    firstTimeSetup = true;
+  }
+  const isYarn = await exists(yarnLockFile);
+
+  // Ensure typescript is installed
+  let ts: typeof import('typescript');
+  try {
+    ts = require(resolve.sync('typescript', {
+      basedir: nodeModules,
+    }));
+  } catch (_) {
+    console.error(
+      chalk.bold.red(
+        `It looks like you're trying to use TypeScript but do not have ${chalk.bold(
+          'typescript',
+        )} installed.`,
+      ),
+    );
+    console.error(
+      chalk.bold(
+        'Please install',
+        chalk.cyan.bold('typescript'),
+        'by running',
+        chalk.cyan.bold(
+          isYarn ? 'yarn add typescript' : 'npm install typescript',
+        ) + '.',
+      ),
+    );
+    console.error(
+      chalk.bold(
+        'If you are not trying to use TypeScript, please remove the ' +
+          chalk.cyan('tsconfig.json') +
+          ' file from your package root (and any TypeScript files).',
+      ),
+    );
+    console.error();
+    process.exit(1);
+    return
+  }
+
+  const compilerOptions: any = {
+    // These are suggested values and will be set when not present in the
+    // tsconfig.json
+    // 'parsedValue' matches the output value from ts.parseJsonConfigFileContent()
+    target: {
+      parsedValue: ts.ScriptTarget.ES5,
+      suggested: 'es5',
+    },
+    lib: { suggested: ['dom', 'dom.iterable', 'esnext'] },
+    allowJs: { suggested: true },
+    skipLibCheck: { suggested: true },
+    esModuleInterop: { suggested: true },
```

Comment:
This should probably be set to `true` instead of having it as a suggestion, [@babel/typescript](https://babeljs.io/docs/en/babel-plugin-transform-typescript) will always have it enabled.

Reply:
Good point, updated